Tests/SwaggerParserTests/SwaggerParserTests.swift

```diff
@@ -1,20 +1,129 @@
+
+fileprivate extension SwaggerParserTests {
+    class func validate(testAllOfBaseSchema schema: ObjectSchema) {
+        validate(thatSchema: schema, named: "TestAllOfBase", hasRequiredProperties: ["base", "test_type"])
+    }
+
+    class func validate(thatSchema schema: ObjectSchema, named name: String, hasRequiredProperties properties: [String]) {
+        XCTAssertEqual(schema.properties.count, properties.count)
+        XCTAssertEqual(schema.required, properties)
+        
+        properties.forEach { property in
+            XCTAssertNotNil(schema.properties.first(where: {$0.key == property}))
+        }
+    }
+    
+    class func validate(thatParameter parameter: Parameter, named parameterName: String, isAnObjectNamed objectName: String, withPropertyName objectPropertyName: String) {
+        guard case .body(_, let schema) = parameter else {
+            XCTFail("\(parameterName) is not a .body."); return
+        }
+        guard case .structure(let structure) = schema else {
+            XCTFail("\(parameterName)'s schema is not a .structure."); return
+        }
+        XCTAssertEqual(structure.name, objectName)
+        
+        guard case .object(let object) = structure.structure else {
+            XCTFail("\(parameterName)'s schema's structure is not an .object."); return
+        }
+        
+        XCTAssertTrue(object.properties.contains(where: {$0.key == objectPropertyName}))
+    }
+    
+    class func validate(thatChildSchema childSchema: Schema, named childName: String, withProperties childProperties: [String], hasParentNamed parentName: String, withProperties parentProperties: [String]) {
+        guard case .allOf(let childAllOf) = childSchema else {
+            XCTFail("\(childName) is not an allOf."); return
+        }
+        XCTAssertEqual(childAllOf.schemas.count, 2)
+        
+        guard
+            let childsParent = childAllOf.schemas.first,
+            case .structure(let childsParentStructure) = childsParent,
+            childsParentStructure.name == parentName,
+            case .object(let childsParentSchema) = childsParentStructure.structure else {
+                XCTFail("\(childName)'s parent is not a Structure<Schema.object>"); return
+        }
+        SwaggerParserTests.validate(thatSchema: childsParentSchema, named: parentName, hasRequiredProperties: parentProperties)
+        
+        guard
+            let child = childAllOf.schemas.last,
+            case .object(let childSchema) = child else {
+                XCTFail("child is not a Structure<Schema.object>"); return
+        }
+        SwaggerParserTests.validate(thatSchema: childSchema, named: childName, hasRequiredProperties: childProperties)
+    }
+}
+
+/// MARK: Swagger Definitions Extension
+
+fileprivate extension Array where Element == Structure<Schema> {
+    func validate(testAllOfChild name: String, withPropertyNames propertyNames: [String]) {
+        let testAllOfChild = self.first(where: {$0.name == name})!
```

Comment:
Let's not do force unwraps - I like what you were doing earlier with the guard statements where the else has XCTFail